### PR TITLE
Add Python performance benchmark test; Add usage and example and RIOSocket option support for C# benchmark; Update Scala benchmark.

### PR DESCRIPTION
**Pull request by qualiu**
Besides normal spark-submit , there's a convenient way use https://github.com/qualiu/testMobius , cluster examples as following:
d:\msgit\lqmMobius\testMobius\scripts\perf\submit-python-perf-test.bat d:\msgit\lqmMobius\python\perf 1 hdfs:///perf/data/deletions/deletions.csv-00000-of-00020
D:\msgit\lqmMobius\testMobius\scripts\perf\submit-scala-perf-test.bat  d:\msgit\lqmMobius\scala\perf  1 hdfs:///perf/data/deletions/deletions.csv-00000-of-00020
D:\msgit\lqmMobius\testMobius\scripts\perf\submit-mobius-perf-test.bat d:\msgit\lqmMobius\csharp\perf 1 hdfs:///perf/data/deletions/deletions.csv-00000-of-00020

### CODE REVIEW

**Comment by qualiu** [comment]
## Key points to this change
* Add Python version benchmark.
* Add a brief overall "Median time costs" report line for 3 versions.
* Add time info to the report line head.

## Minor:
* Add usage and examples for 3 versions.
* Correct invalid website links in C# and Scala version codes.
* Update hard code Scala version to the already defined variable "scala.binary.version"

**Comment by qualiu** [comment]
## Key points to this change
* Add Python version benchmark.
* Add a brief overall "Median time costs" report line for 3 versions
* Add time info to the report line head.
* Add RioSocket option support for C# version benchmark.

## Minor:
* Add usage and examples for 3 versions.
* Correct invalid website links in C# and Scala version codes.
* Update hard code Scala version to the already defined variable "scala.binary.version"

File: csharp/Perf/Microsoft.Spark.CSharp/PerfBenchmark.csproj [hunk 1]
```diff
@@ -60,6 +71,10 @@
   </ItemGroup>
   <ItemGroup />
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <PropertyGroup>
+    <PostBuildEvent Condition=" '$(OS)' == 'Windows_NT' ">copy /y $(ProjectDir)..\..\..\csharp\Worker\Microsoft.Spark.CSharp\bin\$(ConfigurationName)\CSharpWorker.* $(TargetDir)</PostBuildEvent>
```
Comment (skaarthik) [inline csharp/Perf/Microsoft.Spark.CSharp/PerfBenchmark.csproj]: what is this for?

File: csharp/Perf/Microsoft.Spark.CSharp/Program.cs [hunk 2]
```diff
@@ -24,6 +25,15 @@ class PerfBenchmark
 
         public static void Main(string[] args)
         {
+            if (args.Length != 3)
+            {
+                var exe = System.Reflection.Assembly.GetEntryAssembly().Location;
+                Console.WriteLine(@"Usage   : {0}  spark-local-dir   run-count  data-path", exe);
+                Console.WriteLine(@"Example : {0}  D:\Temp\perfTest  10         hdfs:///perfdata/freebasedeletions/*", exe);
```
Comment (skaarthik) [inline csharp/Perf/Microsoft.Spark.CSharp/Program.cs]: Path in format "file:///" is also supported, right?

File: csharp/Perf/Microsoft.Spark.CSharp/PerfBenchmark.csproj (hunk 1 above)
Comment (qualiu) [inline csharp/Perf/Microsoft.Spark.CSharp/PerfBenchmark.csproj]: Add(Copy) the `CSharpWorker.exe.config` to make it possible for the user modify the log level info if they want.

Comment (qualiu) [inline csharp/Perf/Microsoft.Spark.CSharp/PerfBenchmark.csproj]: Add(Copy) the `CSharpWorker.exe.config` to make it possible for the users to modify the log level of worker if they want.

File: csharp/Perf/Microsoft.Spark.CSharp/Program.cs (hunk 2 above)
Comment (qualiu) [inline csharp/Perf/Microsoft.Spark.CSharp/Program.cs]: Yes, such as `file:///d:/mobius/deletions/deletions.csv-00000-of-00020`.
I just give 2 cluster examples of hdfs file.
Do you want me to add a case of this format data path?

Comment (skaarthik) [inline csharp/Perf/Microsoft.Spark.CSharp/Program.cs]: You may want to add that as a supported data path format

File: csharp/Perf/Microsoft.Spark.CSharp/PerfBenchmark.csproj (hunk 1 above)
Comment (skaarthik) [inline csharp/Perf/Microsoft.Spark.CSharp/PerfBenchmark.csproj]: ok

File: csharp/Perf/Microsoft.Spark.CSharp/Program.cs (hunk 2 above)
Comment (qualiu) [inline csharp/Perf/Microsoft.Spark.CSharp/Program.cs]: OK, I've added 2 more examples of more path formats.

**Pull request merged**